File: rb/features.rb
# Nekosbest SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module NekosbestFeatures
  def self.make_feature(name)
    case name
    when "base"
      NekosbestBaseFeature.new
    when "test"
      NekosbestTestFeature.new
    else
      NekosbestBaseFeature.new
    end
  end
end

File: py/nekosbest_sdk/features.py
# Nekosbest SDK feature factory

from nekosbest_sdk.feature.base_feature import NekosbestBaseFeature
from nekosbest_sdk.feature.test_feature import NekosbestTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NekosbestBaseFeature(),
        "test": lambda: NekosbestTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
